src/components/checkbox/checkbox.ts

```diff
@@ -0,0 +1,246 @@
+ */
+enum UiTransitionState {
+  /** The initial state of the component before any user interaction. */
+  Init,
+  /** The state representing the component when it's becoming checked. */
+  Checked,
+  /** The state representing the component when it's becoming unchecked. */
+  Unchecked,
+  /** The state representing the component when it's becoming indeterminate. */
+  Indeterminate
+}
+
+/**
+ * A material design checkbox component. Supports all of the functionality of an HTML5 checkbox,
+ * and exposes a similar API. An MdCheckbox can be either checked, unchecked, indeterminate, or
+ * disabled. Note that all additional accessibility attributes are taken care of by the component,
+ * so there is no need to provide them yourself. However, if you want to omit a label and still
+ * have the checkbox be accessible, you may supply an [aria-label] input.
+ * See: https://www.google.com/design/spec/components/selection-controls.html#selection-controls-checkbox
+ */
+@Component({
+  selector: 'md-checkbox',
+  templateUrl: './components/checkbox/checkbox.html',
+  styleUrls: ['./components/checkbox/checkbox.css'],
+  host: {
+    '[id]': 'id',
+    '[class.md-checkbox]': 'true',
+    '[class.md-checkbox-indeterminate]': 'indeterminate',
+    '[class.md-checkbox-checked]': 'checked',
+    '[class.md-checkbox-disabled]': 'disabled',
+    '[class.md-checkbox-align-end]': 'align == "end"',
+    '[attr.role]': '"checkbox"',
+    '[attr.tabindex]': 'disabled ? -1 : tabindex',
+    '[attr.aria-label]': 'ariaLabel',
+    '[attr.aria-labelledby]': 'labelId',
+    '[attr.aria-checked]': 'getAriaChecked()',
+    '[attr.aria-disabled]': 'disabled',
+    '(click)': 'toggle($event)',
+    '(keyup.space)': 'toggle($event)',
+    '(blur)': 'onTouched()'
+  },
+  providers: [MD_CHECKBOX_CONTROL_VALUE_ACCESSOR],
+  encapsulation: ViewEncapsulation.None
+})
+export class MdCheckbox implements ControlValueAccessor {
+  /**
+   * Attached to the aria-label attribute of the host element. In most cases, arial-labelledby will
+   * take precedence so this may be omitted.
+   */
+  @Input('aria-label') ariaLabel: string = '';
+
+  /** A unique id for the checkbox. If one is not supplied, it is auto-generated. */
+  @Input() id: string = `md-checkbox-${++nextId}`;
+
+  /** Whether or not the checkbox should come before or after the label */
+  @Input() @OneOf(['start', 'end']) align: string = 'start';
+
+  /**
+   * True when the checkbox is disabled, false otherwise. When the checkbox is disabled it cannot
+   * be interacted with. The correct ARIA attributes are applied to denote this to assistive
+   * technology.
+   */
+  @Input() disabled: boolean = false;
+
+  /**
+   * The tabindex attribute for the checkbox. Note that when the checkbox is disabled, the attribute
+   * on the host element will be set to -1, regardless of the actual tabindex value.
+   */
+  @Input() tabindex: number = 0;
```

Comment:
Huh, I remember this working at some point, but it seems to not be the case now. Nevermind,